src/js/actions/export.js

```diff
@@ -533,6 +534,31 @@ define(function (require, exports) {
     setAllNonABLayersExportEnabled.transfers = [setLayerExportEnabled];
 
     /**
+     * Prompt the user to choose a folder by opening an OS dialog.
+     * Keyboard policies are temporarily disabled while the dialog is open.
+     *
+     * @return {Promise.<?string>} Promise of a File Path of the chosen folder, returns null if user-canceled
+     */
+    var promptForFolder = function () {
+        return this.transfer(policyActions.disableKeyboardPolicies)
+            .bind(this)
+            .then(
+                function () {
+                    return _exportService.promptForFolder(_lastFolderPath || "~");
+                },
+                function (e) {
+                    throw new Error("Failed to stash keyboard policies prior to opening OS dialog: " + e.message);
+                }
+            )
+            .finally(function () {
+                return this.transfer(policyActions.reenabledKeyboardPolicies);
+            });
+    };
+    promptForFolder.reads = [];
+    promptForFolder.writes = locks.ALL_LOCKS;
```

Comment:
This makes me a little uncomfortable, but I guess I can't think either of a problem it will cause or a "better" solution. It does seem like if keyboard policies are disabled we shouldn't let other actions run concurrently. So, I guess :+1: